projectroles/utils.py

```diff
@@ -95,3 +109,197 @@ def get_app_names():
             else:
                 ret.append(s[0])
     return sorted(ret)
+
+
+class SidebarContent:
+    """Class for generating sidebar content for the site"""
+
+    def _is_active(self, request, link_names=None):
+        """Check if current URL is active in the sidebar."""
+        # HACK: Avoid circular import
+        from projectroles.urls import urlpatterns
+
+        if request.resolver_match.app_name != 'projectroles':
+            return False
+        url_name = request.resolver_match.url_name
+        if url_name in [u.name for u in urlpatterns]:
+            if link_names:
+                if not isinstance(link_names, list):
+                    link_names = [link_names]
+                if url_name not in link_names:
+                    return False
+            return True
+        return False
+
+    def _is_active_plugin(self, app_plugin, app_name, url_name):
+        """
+        Check if current URL is active in the sidebar for a specific app plugin.
+        """
+        if app_plugin.name.startswith(app_name) and url_name in [
+            u.name for u in getattr(app_plugin, 'urls', [])
+        ]:
+            return True
+        # HACK for remote site views, see issue #1336
+        if (
+            app_name == 'projectroles'
+            and app_plugin.name == 'remotesites'
+            and url_name.startswith('remote_')
+        ):
+            return True
+        return False
+
+    def _is_app_visible(self, plugin, project, user):
+        """Check if app should be visible for user in a specific project."""
+        can_view_app = user.has_perm(plugin.app_permission, project)
+        app_hidden = False
+        if plugin.name in getattr(
+            settings, 'PROJECTROLES_HIDE_PROJECT_APPS', []
+        ):
+            app_hidden = True
+        if (
+            can_view_app
+            and not app_hidden
+            and (
+                project.type == SODAR_CONSTANTS['PROJECT_TYPE_PROJECT']
```

Comment:
Hint: you can define "shortcuts" like `PROJECT_TYPE_PROJECT = SODAR_CONSTANTS['PROJECT_TYPE_PROJECT']` in the module root to shorten constant checks like this nicely.